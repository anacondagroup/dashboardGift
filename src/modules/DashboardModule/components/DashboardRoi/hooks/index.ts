export * from './useRoiDashboardProtector';
export * from './useRoiTable/useRoiTable';
