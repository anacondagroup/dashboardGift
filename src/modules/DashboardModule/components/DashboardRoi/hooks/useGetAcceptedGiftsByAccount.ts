import { TAcceptedGiftByEmailDomain, useGetAcceptedGiftsByAccountQuery } from '@alycecom/services';

import { transformRoiFilters } from '../utils/transformRoiFilters';

import { TDebouncedFilters } from './useRoiTable/useRoiTable';
import { useWaitValidFilters } from './useWaitValidFilters';

export type TUseGetAcceptedGiftsByAccountParams = {
  accountId: string;
  filters: TDebouncedFilters<TAcceptedGiftByEmailDomain>;
};

export const useGetAcceptedGiftsByAccount = ({
  accountId,
  filters,
}: TUseGetAcceptedGiftsByAccountParams): ReturnType<typeof useGetAcceptedGiftsByAccountQuery> & {
  isWaitingForFilters: boolean;
} => {
  const isWaitingForFilters = useWaitValidFilters(filters);
  const queryResult = useGetAcceptedGiftsByAccountQuery(
    {
      filters: transformRoiFilters(filters),
      accountId,
    },
    { skip: isWaitingForFilters },
  );

  return {
    ...queryResult,
    isWaitingForFilters,
  };
};
