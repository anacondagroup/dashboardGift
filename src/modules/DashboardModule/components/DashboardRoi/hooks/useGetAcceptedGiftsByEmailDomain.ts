import { TAcceptedGiftByEmailDomain, useGetAcceptedGiftsByEmailDomainQuery } from '@alycecom/services';

import { transformRoiFilters } from '../utils/transformRoiFilters';

import { TDebouncedFilters } from './useRoiTable/useRoiTable';
import { useWaitValidFilters } from './useWaitValidFilters';

export type TUseGetAcceptedGiftsByEmailDomainParams = {
  emailDomain: string;
  filters: TDebouncedFilters<TAcceptedGiftByEmailDomain>;
};

export const useGetAcceptedGiftsByEmailDomain = ({
  emailDomain,
  filters,
}: TUseGetAcceptedGiftsByEmailDomainParams): ReturnType<typeof useGetAcceptedGiftsByEmailDomainQuery> & {
  isWaitingForFilters: boolean;
} => {
  const isWaitingForFilters = useWaitValidFilters(filters);
  const queryResult = useGetAcceptedGiftsByEmailDomainQuery(
    {
      filters: transformRoiFilters(filters),
      emailDomain,
    },
    { skip: isWaitingForFilters },
  );

  return {
    ...queryResult,
    isWaitingForFilters,
  };
};
