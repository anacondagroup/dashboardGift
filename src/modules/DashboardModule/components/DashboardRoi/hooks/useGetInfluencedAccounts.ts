import { TInfluencedAccounts, useGetInfluencedAccountsQuery } from '@alycecom/services';

import { transformRoiFilters } from '../utils/transformRoiFilters';

import { TDebouncedFilters } from './useRoiTable/useRoiTable';
import { useWaitValidFilters } from './useWaitValidFilters';

export const useGetInfluencedAccounts = (
  filters: TDebouncedFilters<TInfluencedAccounts>,
): ReturnType<typeof useGetInfluencedAccountsQuery> & {
  isWaitingForFilters: boolean;
} => {
  const isWaitingForFilters = useWaitValidFilters(filters);
  const queryResult = useGetInfluencedAccountsQuery(transformRoiFilters(filters), { skip: isWaitingForFilters });

  return {
    ...queryResult,
    isWaitingForFilters,
  };
};
