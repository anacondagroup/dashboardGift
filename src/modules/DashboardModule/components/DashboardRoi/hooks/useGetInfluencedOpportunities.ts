import { TInfluencedOpportunities, useGetInfluencedOpportunitiesQuery } from '@alycecom/services';

import { transformRoiFilters } from '../utils/transformRoiFilters';

import { TDebouncedFilters } from './useRoiTable/useRoiTable';
import { useWaitValidFilters } from './useWaitValidFilters';

export const useGetInfluencedOpportunities = (
  filters: TDebouncedFilters<TInfluencedOpportunities>,
): ReturnType<typeof useGetInfluencedOpportunitiesQuery> & {
  isWaitingForFilters: boolean;
} => {
  const isWaitingForFilters = useWaitValidFilters(filters);
  const queryResult = useGetInfluencedOpportunitiesQuery(transformRoiFilters(filters), { skip: isWaitingForFilters });

  return {
    ...queryResult,
    isWaitingForFilters,
  };
};
