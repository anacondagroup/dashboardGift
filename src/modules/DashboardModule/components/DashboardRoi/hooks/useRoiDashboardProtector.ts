import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';

import {
  getIsSfConnectionStateLoaded,
  getSfConnectionState,
} from '../../../../SettingsModule/store/organisation/integrations/workato/integrations/integrations.selectors';
import { ROI_ROUTES } from '../routePaths';
import { fetchSfConnectionState } from '../../../../SettingsModule/store/organisation/integrations/workato/integrations/integrations.actions';

export const useRoiDashboardProtector = (): {
  isLoaded: boolean;
  isSalesforceConnected: boolean;
  dashboard: string;
} => {
  const dispatch = useDispatch();

  const sfConnectionState = useSelector(getSfConnectionState);
  const isLoaded = useSelector(getIsSfConnectionStateLoaded);

  const dashboard = sfConnectionState === 'connected' ? ROI_ROUTES.FUNNEL : ROI_ROUTES.REPORTING;

  useEffect(() => {
    dispatch(fetchSfConnectionState());
  }, [dispatch]);

  return {
    isLoaded,
    isSalesforceConnected: sfConnectionState === 'connected',
    dashboard,
  };
};
