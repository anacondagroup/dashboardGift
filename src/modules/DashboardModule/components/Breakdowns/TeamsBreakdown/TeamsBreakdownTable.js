import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import { DISPLAY_DATE_FORMAT, TableCellTooltip, TableLoadingLabel } from '@alycecom/ui';
import { TrackEvent } from '@alycecom/services';
import { Link as RouterLink } from 'react-router-dom';
import {
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  TableSortLabel,
  TableFooter,
  TablePagination,
  Typography,
} from '@mui/material';
import moment from 'moment';

import { teamBreakdownShape } from '../../../shapes/teamBreakdown.shape';
import { usePagination } from '../../../../../hooks/usePagination';

export const TeamsBreakdownTableComponent = ({
  renderToolbar,
  items,
  onSort,
  dir,
  sort,
  isLoading,
  linkQueryParams,
  page,
  onPageChange,
}) => {
  const { trackEvent } = TrackEvent.useTrackEvent();
  const rowsPerPage = 10;
  const [teams, showPagination, emptyRows, showEmptyRows] = usePagination(page, rowsPerPage, items, isLoading);

  const pageChangeHandler = useCallback((_, nextPage) => onPageChange(nextPage), [onPageChange]);

  return (
    <>
      {renderToolbar()}
      <Table padding="none">
        <TableHead>
          <TableRow>
            <TableCell>
              <TableSortLabel direction={dir} active={sort === 'name'} onClick={() => onSort('name')}>
                team name
              </TableSortLabel>
            </TableCell>
            <TableCell align="right">
              <TableSortLabel direction={dir} active={sort === 'gifts_sent'} onClick={() => onSort('gifts_sent')}>
                gifts sent
              </TableSortLabel>
            </TableCell>
            <TableCell align="right">
              <TableSortLabel direction={dir} active={sort === 'gifts_viewed'} onClick={() => onSort('gifts_viewed')}>
                gifts viewed
              </TableSortLabel>
            </TableCell>
            <TableCell align="right">
              <TableSortLabel
                direction={dir}
                active={sort === 'gifts_accepted'}
                onClick={() => onSort('gifts_accepted')}
              >
                gifts accepted
              </TableSortLabel>
            </TableCell>
            <TableCell align="right">
              <TableSortLabel
                direction={dir}
                active={sort === 'meetings_booked'}
                onClick={() => onSort('meetings_booked')}
              >
                meeting booked
              </TableSortLabel>
            </TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {teams.map(team => (
            <TableRow key={team.id}>
              <TableCell component="th" scope="row">
                <TableLoadingLabel
                  maxWidth={400}
                  pr={2}
                  isLoading={isLoading}
                  render={() => (
                    <TableCellTooltip
                      renderLabel={() => (
                        <RouterLink
                          onClick={() =>
                            trackEvent('Enterprise dashboard â€” link from teams page to single team view', {
                              team_id: team.id,
                            })
                          }
                          to={`/teams/${team.id}?${linkQueryParams}`}
                        >
                          {team.name}
                          {Boolean(team.archivedAt) && (
                            <Typography display="inline" noWrap color="#99A1BD">
                              &nbsp;(Archived {moment(team.archivedAt).format(DISPLAY_DATE_FORMAT)})
                            </Typography>
                          )}
                        </RouterLink>
                      )}
                      title={team.name}
                    />
                  )}
                />
              </TableCell>
              <TableCell align="right">
                <TableLoadingLabel
                  align="right"
                  pl={2}
                  maxWidth={190}
                  isLoading={isLoading}
                  render={() => team.giftsSent}
                />
              </TableCell>
              <TableCell align="right">
                <TableLoadingLabel
                  align="right"
                  pl={2}
                  maxWidth={190}
                  isLoading={isLoading}
                  render={() => team.giftsViewed}
                />
              </TableCell>
              <TableCell align="right">
                <TableLoadingLabel
                  align="right"
                  pl={2}
                  maxWidth={190}
                  isLoading={isLoading}
                  render={() => team.giftsAccepted}
                />
              </TableCell>
              <TableCell align="right">
                <TableLoadingLabel
                  align="right"
                  pl={2}
                  maxWidth={190}
                  isLoading={isLoading}
                  render={() => team.meetingsBooked}
                />
              </TableCell>
            </TableRow>
          ))}
          {showEmptyRows > 0 && (
            <TableRow style={{ height: 48 * emptyRows }}>
              <TableCell colSpan={12} />
            </TableRow>
          )}
        </TableBody>
        {showPagination && (
          <TableFooter>
            <TableRow>
              <TablePagination
                rowsPerPageOptions={[rowsPerPage]}
                colSpan={12}
                count={items.length}
                rowsPerPage={rowsPerPage}
                page={page}
                SelectProps={{
                  native: true,
                }}
                onPageChange={pageChangeHandler}
              />
            </TableRow>
          </TableFooter>
        )}
      </Table>
    </>
  );
};

TeamsBreakdownTableComponent.propTypes = {
  sort: PropTypes.string.isRequired,
  dir: PropTypes.string.isRequired,
  renderToolbar: PropTypes.func.isRequired,
  onSort: PropTypes.func.isRequired,
  items: PropTypes.arrayOf(teamBreakdownShape).isRequired,
  onPageChange: PropTypes.func.isRequired,
  page: PropTypes.number.isRequired,
  isLoading: PropTypes.bool,
  linkQueryParams: PropTypes.string,
};

TeamsBreakdownTableComponent.defaultProps = {
  linkQueryParams: '',
  isLoading: true,
};

export default TeamsBreakdownTableComponent;
