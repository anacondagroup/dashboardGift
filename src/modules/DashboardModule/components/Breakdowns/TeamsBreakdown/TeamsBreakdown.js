import React, { memo, useCallback } from 'react';
import { SearchField, DateRangeSelect } from '@alycecom/ui';
import PropTypes from 'prop-types';
import { updateSearch } from '@alycecom/modules';
import { Box, Grid } from '@mui/material';
import { makeStyles } from '@mui/styles';
import { useUrlQuery, useSetUrlQuery } from '@alycecom/hooks';

import { teamBreakdownShape } from '../../../shapes/teamBreakdown.shape';

import TeamsBreakdownTable from './TeamsBreakdownTable';

const useStyles = makeStyles(() => ({
  dateRangeFilter: {
    flexGrow: 1,
  },
}));

const TeamsBreakdown = ({ breakdown, isLoading, onChangeDateRange }) => {
  const classes = useStyles();

  const {
    teamsSort = 'name',
    teamsDirection = 'asc',
    teamsSearch = '',
    teamsPage = '0',
    dateRangeFrom = '',
    dateRangeTo = '',
  } = useUrlQuery(['teamsSort', 'teamsDirection', 'teamsSearch', 'teamsPage', 'dateRangeFrom', 'dateRangeTo']);
  const updateUrl = useSetUrlQuery();

  const handleSort = useCallback(
    column => {
      updateUrl({
        teamsSort: column,
        teamsDirection: teamsSort === column && teamsDirection === 'desc' ? 'asc' : 'desc',
        teamsPage: 0,
      });
    },
    [teamsSort, teamsDirection, updateUrl],
  );

  const onPageChange = useCallback(page => updateUrl({ teamsPage: page }), [updateUrl]);

  const onSearch = useCallback(event => updateUrl({ teamsSearch: event.target.value, teamsPage: 0 }), [updateUrl]);

  return (
    <TeamsBreakdownTable
      isLoading={isLoading}
      items={breakdown}
      sort={teamsSort}
      dir={teamsDirection}
      onSort={handleSort}
      page={Number(teamsPage)}
      onPageChange={onPageChange}
      linkQueryParams={updateSearch('', { dateRangeFrom, dateRangeTo })}
      renderToolbar={() => (
        <Grid container direction="row" wrap="nowrap" xs={12}>
          <Grid item xs={10}>
            <SearchField placeholder="Search teams" value={teamsSearch} onChange={onSearch} />
          </Grid>
          <Grid item xs={2}>
            <Box ml={1} display="flex">
              <DateRangeSelect
                className={classes.dateRangeFilter}
                disabled={isLoading}
                from={dateRangeFrom}
                to={dateRangeTo}
                onChange={onChangeDateRange}
              />
            </Box>
          </Grid>
        </Grid>
      )}
    />
  );
};

TeamsBreakdown.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  breakdown: PropTypes.arrayOf(teamBreakdownShape).isRequired,
  onChangeDateRange: PropTypes.func.isRequired,
};

export default memo(TeamsBreakdown);
