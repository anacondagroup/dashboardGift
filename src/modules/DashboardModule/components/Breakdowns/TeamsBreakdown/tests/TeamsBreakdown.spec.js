import React from 'react';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';
import { ThemeProvider, StyledEngineProvider } from '@mui/material';
import { theme } from '@alycecom/ui';
import { updateSearch } from '@alycecom/modules';
import { useUrlQuery } from '@alycecom/hooks';

import TeamsBreakdown from '../TeamsBreakdown';

jest.mock('@mui/material/Grid', () => 'Grid');
jest.mock('../TeamsBreakdownTable', () => ({ renderToolbar, ...props }) => <div {...props}>{renderToolbar()}</div>);
jest.mock('@alycecom/modules/dist/services/url.service');
jest.mock('@alycecom/hooks', () => ({
  useUrlQuery: jest.fn(),
  useSetUrlQuery: jest.fn(),
}));

describe('TeamsBreakdown', () => {
  let props;
  beforeEach(() => {
    props = {
      breakdown: [],
      isLoading: true,
      onChangeDateRange: jest.fn(),
    };

    useUrlQuery.mockReset();
    updateSearch.mockReset();

    updateSearch.mockImplementation((search, updateObj) => JSON.stringify({ search, updateObj }));
  });

  test('render while breakdown is loading', () => {
    useUrlQuery.mockReturnValue({});

    const tree = create(
      shallow(
        <StyledEngineProvider injectFirst>
          <ThemeProvider theme={theme}>
            <TeamsBreakdown {...props} />
          </ThemeProvider>
        </StyledEngineProvider>,
      ),
    );

    expect(tree).toMatchSnapshot();
  });

  test('render with breakdown', () => {
    useUrlQuery.mockReturnValue({
      teamsDirection: 'asc',
      teamsSort: 'someProperty',
      teamsSearch: 'search',
      dateRangeFrom: '2011-11-11',
      dateRangeTo: '2012-12-12',
      teamsPage: '0',
    });
    props = {
      breakdown: [{ id: 0, prop: 'value' }],
      isLoading: false,
      onChangeDateRange: jest.fn(),
    };

    const tree = create(
      shallow(
        <StyledEngineProvider injectFirst>
          <ThemeProvider theme={theme}>
            <TeamsBreakdown {...props} />
          </ThemeProvider>
        </StyledEngineProvider>,
      ),
    );

    expect(tree).toMatchSnapshot();
  });
});
