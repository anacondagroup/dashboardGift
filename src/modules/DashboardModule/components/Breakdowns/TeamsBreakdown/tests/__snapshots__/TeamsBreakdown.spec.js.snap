// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamsBreakdown render while breakdown is loading 1`] = `
<div
  dir="desc"
  isLoading={true}
  items={Array []}
  linkQueryParams="{\\"search\\":\\"\\",\\"updateObj\\":{\\"dateRangeFrom\\":\\"\\",\\"dateRangeTo\\":\\"\\"}}"
  onPageChange={[Function]}
  onSort={[Function]}
  page={0}
  sort=""
>
  <Grid
    container={true}
    direction="row"
    wrap="nowrap"
    xs={12}
  >
    <Grid
      item={true}
      xs={10}
    >
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1dumpt7-MuiFormControl-root-MuiTextField-root"
      >
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart css-xqzzw5-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <div
            className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
          >
            <span
              className="notranslate"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-search  css-1boqai1"
              data-icon="search"
              data-prefix="fas"
              focusable="false"
              onClick={[Function]}
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </div>
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1to944p-MuiInputBase-input-MuiOutlinedInput-input"
            data-testid="search-teams"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Search teams"
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-nnbavb"
            >
              <span
                className="notranslate"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
    </Grid>
    <Grid
      item={true}
      xs={2}
    >
      <div
        className="MuiBox-root css-7i6r7n"
      >
        <div
          className="MuiFormControl-root makeStyles-dateRangeFilter-1 css-uo9nmu-MuiFormControl-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1szc4y1-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={true}
            htmlFor="date-range-select"
          >
            Viewing
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl css-1rpyuhh-MuiInputBase-root-MuiOutlinedInput-root"
            data-testid="DateRangeSelect"
            onClick={[Function]}
          >
            <div
              aria-disabled="true"
              aria-expanded="false"
              aria-haspopup="listbox"
              className="MuiSelect-select MuiSelect-outlined Mui-disabled MuiOutlinedInput-input MuiInputBase-input Mui-disabled css-kath9a-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={null}
              role="button"
              tabIndex={null}
            >
              All Times
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              disabled={true}
              id="date-range-select"
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={false}
              tabIndex={-1}
              value={
                Object {
                  "custom": false,
                  "empty": true,
                  "label": "All Times",
                  "value": Object {},
                }
              }
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined Mui-disabled css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1z7n62"
              >
                <span>
                  Viewing
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </Grid>
  </Grid>
</div>
`;

exports[`TeamsBreakdown render with breakdown 1`] = `
<div
  dir="asc"
  isLoading={false}
  items={
    Array [
      Object {
        "id": 0,
        "prop": "value",
      },
    ]
  }
  linkQueryParams="{\\"search\\":\\"\\",\\"updateObj\\":{\\"dateRangeFrom\\":\\"2011-11-11\\",\\"dateRangeTo\\":\\"2012-12-12\\"}}"
  onPageChange={[Function]}
  onSort={[Function]}
  page={0}
  sort="someProperty"
>
  <Grid
    container={true}
    direction="row"
    wrap="nowrap"
    xs={12}
  >
    <Grid
      item={true}
      xs={10}
    >
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1dumpt7-MuiFormControl-root-MuiTextField-root"
      >
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd css-kgdphc-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <div
            className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
          >
            <span
              className="notranslate"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-search  css-1boqai1"
              data-icon="search"
              data-prefix="fas"
              focusable="false"
              onClick={[Function]}
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </div>
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-tr2z7b-MuiInputBase-input-MuiOutlinedInput-input"
            data-testid="search-teams"
            disabled={false}
            id="mui-2"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Search teams"
            required={false}
            type="text"
            value="search"
          />
          <div
            className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          >
            <div
              className="MuiBox-root css-rcmis8"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-times Text-Pointer css-1hbp0r0"
                data-icon="times"
                data-prefix="fas"
                focusable="false"
                onClick={[Function]}
                role="img"
                style={Object {}}
                viewBox="0 0 352 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-nnbavb"
            >
              <span
                className="notranslate"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
    </Grid>
    <Grid
      item={true}
      xs={2}
    >
      <div
        className="MuiBox-root css-7i6r7n"
      >
        <div
          className="MuiFormControl-root makeStyles-dateRangeFilter-1 css-uo9nmu-MuiFormControl-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1szc4y1-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={true}
            htmlFor="date-range-select"
          >
            Viewing
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1rpyuhh-MuiInputBase-root-MuiOutlinedInput-root"
            data-testid="DateRangeSelect"
            onClick={[Function]}
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-kath9a-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              
          Nov 11, 2011
           - 
          Dec 12, 2012
        
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              disabled={false}
              id="date-range-select"
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={false}
              tabIndex={-1}
              value={
                Object {
                  "custom": true,
                  "label": "Custom Range",
                  "value": Object {
                    "from": "",
                    "to": "",
                  },
                }
              }
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1z7n62"
              >
                <span>
                  Viewing
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </Grid>
  </Grid>
</div>
`;
